engine/src/main/java/com/datatorrent/stram/cli/ApexCli.java

```diff
@@ -3516,6 +3517,16 @@ private void checkPlatformCompatible(AppPackage ap)
     }
   }
 
+  private void checkJavaVersionCompatible(AppPackage ap)
+  {
+    String buildJdkVersion = ap.getBuildJdkVersion();
+    String systemVersion = System.getProperty("java.version");
+    if (VersionInfo.compare(buildJdkVersion, systemVersion) > 0) {
```

Comment:
I wasn't sure about comparing the Build-jdk version with the current Java version (the one running the Apex cli). Note that one can use Build-jdk version 1.8 to generate class files of version 1.7 (class version major 51) and if there are no other issues it should ideally run. But this check will catch such cases, preventing running of such packages. However the Java runtime libraries could still present a problem so in my opinion this check is good for now.